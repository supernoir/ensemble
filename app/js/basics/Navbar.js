import React from 'react';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import { Menu, Container, Dropdown, Icon } from 'semantic-ui-react';
import LanguageSwitcher from './LanguageSwitcher';

export default class Navbar extends React.Component {

	render() {
		return (
			<Menu inverted fixed="top">
				<Container>
					<Menu.Item as="a" header>
						<Link to="/">
							{intl.get('brand')}
						</Link>
					</Menu.Item>
					<Menu.Item as="a" header><Link to="/projects"><Icon circular inverted name="book" /> {intl.get('entity.projects')}</Link></Menu.Item>
					<Menu.Item as="a" header><Link to="/characters"><Icon circular inverted name="address card" /> {intl.get('entity.characters')}</Link></Menu.Item>
					<Menu.Item as="a" header><Link to="/tags"><Icon circular inverted name="tags" /> {intl.get('entity.tags')}</Link></Menu.Item>
					{/**
						<Menu.Item disabled header><Link to="/journeys"><Icon circular inverted name="map signs" /> {intl.get('entity.journeys')}</Link></Menu.Item>
						<Menu.Item disabled header><Link to="/events"><Icon circular inverted name="time" /> {intl.get('entity.timeline')}</Link></Menu.Item>
					*/}
					<Menu.Item as="a" header><Link to="/admin"><Icon circular inverted name="cogs" /> {'Admin Panel'}</Link></Menu.Item>

					<Dropdown
						item
						simple
						icon="user"
						text={'supernoir'}
					>
						<Dropdown.Menu>
							<Dropdown.Item href='/profile'>{'My Profile'}</Dropdown.Item>
							<Dropdown.Divider />
							<Dropdown.Item href='/login'>
								{'Log out'}
							</Dropdown.Item>
						</Dropdown.Menu>
					</Dropdown>
					<LanguageSwitcher setCurrentLocale={(locale)=>{this.props.setCurrentLocale(locale);}} {...this.props}/>
				</Container>
			</Menu>
		);
	}
}
