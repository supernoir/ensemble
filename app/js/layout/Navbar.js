import React from 'react';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import { Menu, Container, Dropdown, Icon } from 'semantic-ui-react';

export default class Navbar extends React.Component {

	render() {
		return (
			<Menu inverted fixed="top">
				<Container>
					<Menu.Item as="a" header>
						<Link to="/">
							{intl.get('brand')}
						</Link>
					</Menu.Item>
					<Menu.Item as="a" header><Link to="/books"><Icon circular inverted name="book" /> {intl.get('entity.books')}</Link></Menu.Item>
					<Menu.Item as="a" header><Link to="/characters"><Icon circular inverted name="address card" /> {intl.get('entity.characters')}</Link></Menu.Item>
					<Menu.Item disabled header><Link to="/journeys"><Icon circular inverted name="map signs" /> {intl.get('entity.journeys')}</Link></Menu.Item>
					<Menu.Item disabled header><Link to="/events"><Icon circular inverted name="time" /> {intl.get('entity.timeline')}</Link></Menu.Item>

					<Dropdown
						item
						simple
						icon="user"
						text={'supernoir'}
					>
						<Dropdown.Menu>
							<Dropdown.Item href='/profile'>{'My Profile'}</Dropdown.Item>
							<Dropdown.Divider />
							<Dropdown.Item href='/login'>
								{'Log out'}
							</Dropdown.Item>
						</Dropdown.Menu>
					</Dropdown>
				</Container>
			</Menu>
		);
	}
}
