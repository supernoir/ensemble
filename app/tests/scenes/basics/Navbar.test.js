import React from 'react';
import { shallow } from 'enzyme';

const testData = {
	setCurrentLocale: jest.fn(),
};

// Import Components
import Navbar from '../../../js/basics/Navbar';

test('Navbar -- Snapshot Test', () => {
	const wrapper = shallow(<Navbar
		close={() => testData.setCurrentLocale()}
	/>
	);
	expect(wrapper).toMatchSnapshot();
});
