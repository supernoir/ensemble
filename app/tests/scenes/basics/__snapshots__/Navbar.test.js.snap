// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar -- Snapshot Test 1`] = `
<Menu
  fixed="top"
  inverted={true}
>
  <Container>
    <MenuItem
      header={true}
    >
      <Link
        replace={false}
        to="/"
      />
    </MenuItem>
    <MenuItem
      header={true}
    >
      <Link
        replace={false}
        to="/projects"
      >
        <Icon
          as="i"
          circular={true}
          inverted={true}
          name="book"
        />
         
      </Link>
    </MenuItem>
    <MenuItem
      header={true}
    >
      <Link
        replace={false}
        to="/characters"
      >
        <Icon
          as="i"
          circular={true}
          inverted={true}
          name="address card"
        />
         
      </Link>
    </MenuItem>
    <MenuItem
      header={true}
    >
      <Link
        replace={false}
        to="/tags"
      >
        <Icon
          as="i"
          circular={true}
          inverted={true}
          name="tags"
        />
         
      </Link>
    </MenuItem>
    <MenuItem
      header={true}
    >
      <Link
        replace={false}
        to="/admin"
      >
        <Icon
          as="i"
          circular={true}
          inverted={true}
          name="cogs"
        />
         
        Admin Panel
      </Link>
    </MenuItem>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      closeOnEscape={true}
      deburr={false}
      icon="user"
      item={true}
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      simple={true}
      text="supernoir"
      wrapSelection={true}
    >
      <DropdownMenu>
        <DropdownItem
          href="/profile"
        >
          My Profile
        </DropdownItem>
        <DropdownDivider />
        <DropdownItem
          href="/login"
        >
          Log out
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
    <LanguageSwitcher
      close={[Function]}
      setCurrentLocale={[Function]}
    />
  </Container>
</Menu>
`;
